gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsService.java

```diff
@@ -0,0 +1,508 @@
+      return new ZoneListOptions(DnsServiceRpc.Option.FIELDS, ZoneField.selector(fields));
+    }
+
+    /**
+     * Returns an option to specify a page token.
+     *
+     * <p>The page token (returned from a previous call to list) indicates from where listing should
+     * continue.
+     */
+    public static ZoneListOptions pageToken(String pageToken) {
+      return new ZoneListOptions(DnsServiceRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * The maximum number of zones to return per RPC.
+     *
+     * <p>The server can return fewer zones than requested. When there are more results than the
+     * page size, the server will return a page token that can be used to fetch other results.
+     */
+    public static ZoneListOptions pageSize(int pageSize) {
+      return new ZoneListOptions(DnsServiceRpc.Option.PAGE_SIZE, pageSize);
+    }
+
+    /**
+     * Restricts the list to return only zones with this fully qualified domain name.
+     */
+    public static ZoneListOptions dnsName(String dnsName) {
+      return new ZoneListOptions(DnsServiceRpc.Option.DNS_NAME, dnsName);
+    }
+  }
+
+  /**
+   * Class for specifying project options.
+   */
+  class ProjectOptions extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 6817937338218847748L;
+
+    ProjectOptions(DnsServiceRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the project's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all project fields are returned. {@code
+     * ProjectOptions.fields} can be used to specify only the fields of interest. Project ID is
+     * always returned, even if not specified. {@link ProjectField} provides a list of fields that
+     * can be used.
+     */
+    public static ProjectOptions fields(ProjectField... fields) {
+      return new ProjectOptions(DnsServiceRpc.Option.FIELDS, ProjectField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying change request field options.
+   */
+  class ChangeRequestFieldOptions extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 1067273695061077782L;
+
+    ChangeRequestFieldOptions(DnsServiceRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify which fields of DNS records to be added by the {@link
+     * ChangeRequest} should be returned by the service.
+     *
+     * <p>If this option is not provided, all record fields are returned. {@code
+     * ChangeRequestFieldOptions.additionsFields} can be used to specify only the fields of
+     * interest. The name of the DNS record always returned, even if not specified. {@link
+     * DnsRecordField} provides a list of fields that can be used.
+     */
+    public static ChangeRequestFieldOptions additionsFields(DnsRecordField... fields) {
```

Comment:
I think this needs to be part of the ChangeRequestField (same for deletionsFields)